ikarus/utils/observer/nonlinearsolverlogger.cpp

```diff
@@ -8,48 +8,37 @@
 #pragma GCC diagnostic ignored "-Wswitch-enum"
 
 namespace Ikarus {
-
-void NonLinearSolverLogger::updateImpl(NonLinearSolverMessages message) {
+void NonLinearSolverLogger::updateImpl(NonLinearSolverMessages message, const NonLinearSolverState& state) {
   switch (message) {
-    case NonLinearSolverMessages::ITERATION_STARTED:
-      break;
     case NonLinearSolverMessages::INIT:
-      iters_ = 1;
-      rNorm_ = 0.0;
-      dNorm_ = 0.0;
       spdlog::info("Non-linear solver started:");
-      spdlog::info("{:<11} {:<20} {:<20} {:<20}", "Ite", "normR", "normD", "lambda");
-      spdlog::info("-------------------------------------------------------------------------------");
+      if (state.lambda and state.subsidiaryFunction)
+        spdlog::info("{:>4} {:>17} {:>19} {:>23} {:>15}", "Ite", "ResidualNorm", "CorrectionNorm", "SubsidiaryFunction",
+                     "LoadFactor");
+      else
+        spdlog::info("{:>4} {:>17} {:>19}", "Ite", "ResidualNorm", "CorrectionNorm");
+      spdlog::info("----------------------------------------------------------------------------------");
       break;
-    case NonLinearSolverMessages::ITERATION_ENDED:
-      spdlog::info("{} {:<10d} {:<20.2e} {:<20.2e} {:<20.2e}", "", iters_, rNorm_, dNorm_, lambda_);
-      ++iters_;
+    case NonLinearSolverMessages::ITERATION_STARTED:
       break;
-    default:
+    case NonLinearSolverMessages::ITERATION_ENDED:
+      if (state.lambda and state.subsidiaryFunction)
+        spdlog::info("{} {:>3d} {:>17.3e} {:>19.3e} {:>23.3e} {:>15.3e}", "", state.currentIter, state.residualNorm,
+                     state.correctionNorm, state.subsidiaryFunction.value(), state.lambda.value());
+      else
+        spdlog::info("{} {:>3d} {:>17.3e} {:>19.3e}", "", state.currentIter, state.residualNorm, state.correctionNorm);
       break;
-  }
-}
-
-void NonLinearSolverLogger::updateImpl(NonLinearSolverMessages message, double val) {
-  switch (message) {
     case NonLinearSolverMessages::RESIDUALNORM_UPDATED:
```

Comment:
Do you think it is nicer to remove these cases like `RESIDUALNORM_UPDATED` and similar which is doing nothing here in this file? The `default` behavior is similar to these cases anyways